src/ui/components/FocusContextReduxAdapter.tsx

```diff
@@ -12,17 +23,61 @@ import { rangeForFocusRegion } from "ui/utils/timeline";
 // Adapter that reads focus region (from Redux) and passes it to the FocusContext.
 export default function FocusContextReduxAdapter({ children }: PropsWithChildren) {
   const dispatch = useAppDispatch();
+
   const loadedRegions = useAppSelector(getLoadedRegions);
+  const prevLoadingRegionsRef = useRef<TimeStampedPointRange[] | null>(
+    loadedRegions?.loading || null
+  );
+
   const focusRegion = useAppSelector(getFocusRegion);
+  const prevFocusRegionRef = useRef<FocusRegion | null>(focusRegion);
 
   const [isPending, startTransition] = useTransition();
   const [deferredFocusRegion, setDeferredFocusRegion] = useState<FocusRegion | null>(null);
 
   useEffect(() => {
-    startTransition(() => {
-      setDeferredFocusRegion(focusRegion);
-    });
-  }, [focusRegion, loadedRegions]);
+    let newFocusRegion = focusRegion;
+
+    let loading = loadedRegions?.loading || null;
+    let loadingBeginTime = loading?.[0]?.begin.time ?? null;
+    let loadingEndTime = loading?.[0]?.end.time ?? null;
```

Comment:
There will only ever be 1 `loading` range, right? (Wouldn't make sense to break this up, right?)

Reply:
Yep, that is my understanding as well!

Reply:
Good to confirm.